docs/generated/sql/bnf/create_func_stmt.bnf

```diff
@@ -1,2 +1,2 @@
 create_func_stmt ::=
-	'CREATE' ( 'OR' 'REPLACE' |  ) 'FUNCTION' routine_create_name '(' ( ( ( ( routine_param | routine_param   | routine_param   ) ) ( ( ',' ( routine_param | routine_param   | routine_param   ) ) )* ) |  ) ')' 'RETURNS' ( 'SETOF' |  ) routine_return_type ( ( ( ( 'AS' ( 'SCONST' ) ( ',' 'SCONST' |  ) | 'LANGUAGE' 'SQL' | ( 'CALLED' 'ON' 'NULL' 'INPUT' | 'RETURNS' 'NULL' 'ON' 'NULL' 'INPUT' | 'STRICT' | 'IMMUTABLE' | 'STABLE' | 'VOLATILE' | 'LEAKPROOF' | 'NOT' 'LEAKPROOF' ) ) ) ( ( ( 'AS' ( 'SCONST' ) ( ',' 'SCONST' |  ) | 'LANGUAGE' 'SQL' | ( 'CALLED' 'ON' 'NULL' 'INPUT' | 'RETURNS' 'NULL' 'ON' 'NULL' 'INPUT' | 'STRICT' | 'IMMUTABLE' | 'STABLE' | 'VOLATILE' | 'LEAKPROOF' | 'NOT' 'LEAKPROOF' ) ) ) )* ) |  ) opt_routine_body
```

Comment:
I think we should go back to following the `AS` option with a string constant called `routine_body_str` or something.
Also, the `routine_body` (`RETURN ...` or `BEGIN ATOMIC ... END`) syntax isn't implemented: https://github.com/cockroachdb/cockroach/issues/85144.

Both of the above apply to procedures, as well.

Reply:
Updated the CREATE PROCEDURE and CREATE FUNCTION diagrams (as reflected in the new screenshots above).

Reply:
Looks like something happened to the other create function options (volatile, strict etc.).

For CREATE FUNCTION, these options can be added in any order after the RETURN block:
```
//  { LANGUAGE lang_name
//    | { IMMUTABLE | STABLE | VOLATILE }
//    | [ NOT ] LEAKPROOF
//    | { CALLED ON NULL INPUT | RETURNS NULL ON NULL INPUT | STRICT }
//    | AS 'definition'
//  } ...
```
For CREATE PROCEDURE, these options can be added in any order after the parameters:
```
//  { LANGUAGE lang_name
//    | AS 'definition'
//  } ...
```
Note that the `AS` syntax is just another option; it can be interspersed anywhere with the other options.

Reply:
I misunderstood your initial comment as saying that all the volatility/etc. clauses should be contained in a `routine_body_str` after `AS`. But were you just suggesting that we add that constant name after the `AS` in both CREATE FUNCTION and CREATE PROCEDURE? If so, that's easy.

Reply:
Updated -- please see latest screenshots.